include/color_spinor_field.h

```diff
@@ -1054,6 +1056,38 @@ namespace quda
   */
   void spinorDistanceReweight(ColorSpinorField &src, double alpha0, int t0);
 
+  /**
+     @brief Helper function for determining if the spin of the fields is the same.
+     @param[in] a Input field
+     @param[in] b Input field
+     @return If spin is unique return the number of spins
+   */
+  inline int Spin_(const char *func, const char *file, int line, const ColorSpinorField &a, const ColorSpinorField &b)
```

Comment:
Should we add a version of this for vector_ref/cvector_ref/etc of this as well?

Reply:
I would say no.  The `Nspin()` methods of `cvector_ref<ColorSpinorField>` themselves already test internal for consistency.  I don't think we need to go the length of comparisons between arbitrary combinations of fields and cvector_refs of fields.